addons/hr_attendance/models/hr_employee.py

```diff
@@ -20,18 +20,20 @@ class HrEmployeeBase(models.AbstractModel):
     hours_today = fields.Float(compute='_compute_hours_today')
     hours_last_month_display = fields.Char(compute='_compute_hours_last_month')
 
+    @api.depends('user_id.im_status', 'attendance_state')
     def _compute_presence_state(self):
         """
         Override to include checkin/checkout in the presence state
         Attendance has the second highest priority after login
         """
         super()._compute_presence_state()
-        employees = self.filtered(lambda employee: employee.hr_presence_state != 'present')
+        employees = self.filtered(lambda e: e.hr_presence_state != 'present')
+        working_now_list = self._get_employee_working_now()
```

Comment:
As `_get_employee_working_now`seems to be time-consuming, it better to reduce number of records.
```suggestion
       employee_to_check_working = employee.filtered(lambda e: e.attendance_state == 'checked_out' and employee.hr_presence_state == 'to_define')
        working_now_list = employee_to_check_working._get_employee_working_now()
```